geom/src/polyline.rs

```diff
@@ -160,14 +167,29 @@ impl PolyLine {
         self_pts.extend(other_pts.iter().skip(1));
         PolyLine::new(self_pts)
     }
+
+    /// Like `extend`, but panics on failure.
     pub fn must_extend(self, other: PolyLine) -> PolyLine {
         self.extend(other).unwrap()
     }
+
+    /// Extends `self` by a single point. Assumes the last point and this new point are different
+    /// and panics otherwise. Doesn't clean up any intermediate points.
     pub fn must_push(self, pt: Pt2D) -> PolyLine {
         let new = PolyLine::must_new(vec![self.last_pt(), pt]);
         self.must_extend(new)
     }
 
+    /// Like `extend`, but handles the last and first point not matching by inserting that point.
+    /// Doesn't clean up any intermediate points.
+    pub fn force_extend(mut self, other: PolyLine) -> Result<PolyLine> {
+        if *self.pts.last().unwrap() != other.pts[0] {
+            // TODO Blindly... what if we need to do the angle collapsing?
```

Comment:
There's lots of mess accumulating here. :(   It'd be great to document and enforce some invariants on `PolyLines` consistently. Or maybe switch to `geo::LineString` internally, if it can help with some of these.

Reply:
`extend`, `append`, `force_extend`, `must_extend`

It's clear enough when reading the source what the difference is, but from the call site I'm left guessing. Names are really tough. 

What about... `connect`?  

```diff
- line1.force_extend(line2)
+ line1.connect(line2)
```